src/core/memory/mod.rs

```diff
@@ -86,4 +86,12 @@ impl Memory {
             }
         }
     }
+
+    pub fn shift_right(&mut self, register: u16) {
+        // self.ram[register] >> 1;
```

Comment:
This was basically what I had previously done, but it gave compiler error, still need to figure this out.

Reply:
You need to do an assignment to the register: `self.ram[register] = self.ram[register] >> 1`